cloudinit/config/cc_set_hostname.py

```diff
@@ -31,25 +43,28 @@
 data are available locally. This ensures that the desired hostname is applied
 before any DHCP requests are preformed on these platforms where dynamic DNS is
 based on initial hostname.
-
-**Internal name:** ``cc_set_hostname``
-
-**Module frequency:** always
-
-**Supported distros:** all
-
-**Config keys**::
-
-    preserve_hostname: <true/false>
-    prefer_fqdn_over_hostname: <true/false>
-    fqdn: <fqdn>
-    hostname: <fqdn/hostname>
 """
 
-import os
-
-from cloudinit import util
-from cloudinit.atomic_helper import write_json
+meta: MetaSchema = {
+    "id": "cc_set_hostname",
+    "name": "Set Hostname",
+    "title": "Set hostname and FQDN",
+    "description": MODULE_DESCRIPTION,
+    "distros": [ALL_DISTROS],
+    "frequency": frequency,
+    "examples": [
+        "perserve_hostname: true",
+        dedent(
+            """\
+            hostname: myhost
```

Comment:
Isn't a lot of it technically distro-dependent? Also the interaction between this module and update_hostname is still confusing to me and I think it could probably be simplified down into one module...but not here.